src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs

```diff
@@ -87,12 +87,12 @@ static void SetKnownParameterValue(AzureBicepResource r, string key, Func<AzureB
             // HACK: This forces parameters to be resolved for any AzureProvisioningResource
             r.GetBicepTemplateFile();
 
-            // REVIEW: the secret key vault resources aren't coupled to the container app environment. This
-            // is a side effect from how azd worked. We can move them to another service in the future.
+            // This will throw an exception if there's no value specified, in this new mode, we don't no longer support
+            // automagic secret key vault references.
             SetKnownParameterValue(r, AzureBicepResource.KnownParameters.KeyVaultName, environment.GetSecretOutputKeyVault);
 
             // Set the known parameters for the container app environment
-            SetKnownParameterValue(r, AzureBicepResource.KnownParameters.PrincipalId, _ => environment.ManagedIdentityId);
+            SetKnownParameterValue(r, AzureBicepResource.KnownParameters.PrincipalId, _ => environment.ContainerRegistryManagedIdentityId);
             SetKnownParameterValue(r, AzureBicepResource.KnownParameters.PrincipalType, _ => "ServicePrincipal");
             SetKnownParameterValue(r, AzureBicepResource.KnownParameters.PrincipalName, _ => environment.PrincipalName);
```

Comment:
It feels wrong that we still have `PrincipalName` on the environment. Shouldn't it be `ContainerRegistryManagedIdentityPrincipalName`?

Reply:
I almost half wonder if we should no longer support these "KnownParameters" in ACA. Our Azure resources will never use them (or at least, if they do it is a mistake that needs to be fixed).

Reply:
Separate change. Not in this PR, it's too big already 😄 . I agree.